File: beagle-ui/src/main/java/br/com/zup/beagleui/framework/widget/ui/ListView.kt
package br.com.zup.beagleui.framework.widget.ui

import br.com.zup.beagleui.framework.widget.core.Widget

data class ListView(
    val rows: Widget? = null,
    val remoteDataSource: String? = null,
    val loadingState: Widget? = null
) : Widget

File: Sources/BeagleUI/Sources/UIModels/ListView.swift
//
//  ListView.swift
//  BeagleUI
//
//  Created by Daniel Tes on 12/09/19.
//  Copyright © 2019 Daniel Tes. All rights reserved.
//

public struct ListView: Widget {
    
    let rows: [Widget]?
    let remoteDataSource: String?
    let loadingState: Widget?
    let direction: ListDirection
    
    init(
        rows: [Widget]? = nil,
        remoteDataSource: String? = nil,
        loadingState: Widget? = nil,
        direction: ListDirection = .vertical
    ) {
        self.rows = rows
        self.remoteDataSource = remoteDataSource
        self.loadingState = loadingState
        self.direction = direction
    }
}

public enum ListDirection {
    case vertical
    case horizontal
}
